### PR TITLE
fix: set max for max cookie number to `math.MaxInt`

### PR DESCRIPTION
**Pull request by onematchfox**
The `goreleaser`-based release build for [terraform-provider-argocd](https://github.com/oboukili/terraform-provider-argocd) is currently failing with the following error:

```
  ⨯ release failed after 3m53s               error=failed to build for freebsd_arm_6: exit status 1: # github.com/argoproj/argo-cd/v2/util/http
../../../go/pkg/mod/github.com/argoproj/argo-cd/v2@v2.8.3/util/http/http.go:22:77: cannot use math.MaxInt64 (untyped int constant 9223372036854775807) as int value in argument to env.ParseNumFromEnv (overflows)
```

The affected code in this project was introduced in #14979, which, set the max value of `common.EnvMaxCookieNumber` to `math.MaxInt64`. This, as `goreleaser` correctly identifies, results in an overflow since the `max` parameter in `ParseNumFromEnv` is an `int` (not `int64`). See https://github.com/argoproj/argo-cd/blob/6fa9b17eb757d401cfb815271ad4ee1141152219/util/env/env.go#L20

Why `goreleaser` is failing because of an issue initialising code in this project is a bit of a mystery to me, but regardless, this PR addresses the issue (assuming, of course, that `int` is big enough and the code should not be updated to use `env.ParseInt64FromEnv`).

Related: https://github.com/oboukili/terraform-provider-argocd/pull/354

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

**Review by blakepettersson — APPROVED** [review]

**Review by ishitasequeira — APPROVED** [review]
LGTM!

File: util/http/http.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ const (
 
 // max number of chunks a cookie can be broken into. To be compatible with
 // widest range of browsers, you shouldn't create more than 30 cookies per domain
-var maxCookieNumber = env.ParseNumFromEnv(common.EnvMaxCookieNumber, 20, 0, math.MaxInt64)
+var maxCookieNumber = env.ParseNumFromEnv(common.EnvMaxCookieNumber, 20, 0, math.MaxInt32)
```
Comment (gdsoumya) [inline util/http/http.go:33]: We should probably do `math.MaxInt` instead, it should automatically determine the max value depending on system architecture. Also `ParseNumFromEnv` limits the number to be in `(math.MinInt, math.MaxInt)` so this is should safe to do.

```suggestion
var maxCookieNumber = env.ParseNumFromEnv(common.EnvMaxCookieNumber, 20, 0, math.MaxInt)
```

**Review by gdsoumya — CHANGES_REQUESTED** [review]

**Review by gdsoumya — APPROVED** [review]

**Comment by onematchfox** [comment]
Thanks for the quick review folks. @gdsoumya are you able to cherry pick this across to the `2.8` and `2.9` release branches?

**Comment by gdsoumya** [comment]
@onematchfox I think we can do that but I'll still check with other maintainers and see before doing that.

**Comment by gdsoumya** [comment]
@onematchfox the commit has been cherry picked to release branches.

**Comment by onematchfox** [comment]
Thanks!

**Pull request merged**